lib/HAL/Compile/TargetCompilationManager.cpp

```diff
@@ -155,3 +162,65 @@ void TargetCompilationManager::disableTiming() { rootTimer.stop(); }
 mlir::TimingScope TargetCompilationManager::getTimer(llvm::StringRef name) {
   return rootTimer.nest(name);
 }
+
+bool TargetCompilationManager::emitDiagnostics() {
+  using namespace qssc::config;
+  // NOLINTNEXTLINE(misc-const-correctness)
+  bool foundError = false;
+  const DiagList diagnostics(target.getDiagnostics());
+  for (auto &diag : diagnostics) {
+    auto severity = diag.severity;
+    switch (config.getVerbosityLevel()) {
+    case QSSVerbosity::Error:
+      switch (severity) {
+      case Severity::Fatal:
+      case Severity::Error:
+        foundError = true;
+        (void)qssc::emitDiagnostic(diagnosticCb, diag);
+        llvm::errs() << diag.toString() << "\n";
+        break;
+      case Severity::Warning:
+      case Severity::Info:
+        break;
+      default:
+        llvm_unreachable("Unknown diagnostic severity");
+      }
+      break;
+    case QSSVerbosity::Warn:
+      switch (severity) {
+      case Severity::Fatal:
+      case Severity::Error:
+        foundError = true;
+        [[fallthrough]];
+      case Severity::Warning:
+        (void)qssc::emitDiagnostic(diagnosticCb, diag);
+        llvm::errs() << diag.toString() << "\n";
+        break;
+      case Severity::Info:
+        break;
+      default:
+        llvm_unreachable("Unknown diagnostic severity");
+      }
+      break;
+    case QSSVerbosity::Info:
+    case QSSVerbosity::Debug:
```

Comment:
Maybe I'm misunderstanding the logic, but setting `foundError = true;` in the case of `Warn`, `Info` and `Debug` isn't something we wanna do, Right?

Reply:
For this case `foundError` is only set to true in the case `Fatal` or `Error`, then diagnostic would be emitted. If the severity is only Warn then foundError is not changed and only the diagnostic is emitted.

Reply:
Ahh... Gotcha. I misinterpreted. Thank you!